### PR TITLE
Pass Storage by value

### PR DESCRIPTION
**Pull request by smessmer**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#11677 Pass Storage by value**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D9823457/)

-

Differential Revision: [D9823457](https://our.internmc.facebook.com/intern/diff/D9823457/)

### CODE REVIEW

File: aten/src/ATen/core/TensorImpl.cpp [hunk 1]
```diff
@@ -17,28 +17,32 @@ const Tensor& TensorImpl::grad() const {
...
-      is_variable_(is_variable) {}
+TensorImpl::TensorImpl(Storage storage, TensorTypeId type_id, bool is_variable)
+    :storage_(std::move(storage)),
+    storage_offset_(0),
+    sizes_{0},
+    strides_{1},
+    is_contiguous_(true),
+    numel_(0),
+    type_id_(type_id),
+    scalar_type_(storage_ ? dataTypeToScalarType(storage_.dtype().id()) : ScalarType::Undefined),
```
Comment (ezyang) [inline aten/src/ATen/core/TensorImpl.cpp]: So... is referencing `storage_` actually valid here?

**Review by ezyang — CHANGES_REQUESTED** [review]
doesn't pass ci

**Comment by SsnL** [comment]
ping @smessmer

**Comment by soumith** [comment]
looks abandoned

**Pull request closed without merging**